catalog/core/catalog-core-validationparser/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -152,10 +169,79 @@ private void commitStaged(List<Callable<Boolean>> stagedAdds) throws Exception {
 
     @Override
     public void update(File file) throws Exception {
+        undo(file);
+        apply(file);
     }
 
     @Override
     public void uninstall(File file) throws Exception {
+        undo(file);
+    }
+
+    private void undo(File file) {
+        final String fileName = file.getName();
+
+        LOGGER.debug("Reversing the changes applied by file [{}].", fileName);
+
+        List<ServiceRegistration<MetacardType>> metacardTypeServices = metacardTypesByFile.get(
+                fileName);
+        if (metacardTypeServices != null) {
+            metacardTypeServices.forEach(ServiceRegistration::unregister);
+            metacardTypeServices.clear();
+        }
+
+        List<ServiceRegistration<MetacardValidator>> validatorServices =
+                metacardValidatorsByFile.get(fileName);
+        if (validatorServices != null) {
+            validatorServices.forEach(ServiceRegistration::unregister);
+            validatorServices.clear();
+        }
+
+        Set<String> attributes = attributesByFile.get(fileName);
+        if (attributes != null) {
+            attributes.forEach(attributeRegistry::deregisterAttribute);
+            attributes.clear();
+        }
+
+        List<Outer.Default> defaults = defaultsByFile.get(fileName);
+        if (defaults != null) {
+            defaults.forEach(theDefault -> {
+                if (CollectionUtils.isEmpty(theDefault.metacardTypes)) {
+                    defaultAttributeValueRegistry.removeDefaultValue(theDefault.attribute);
+                } else {
+                    theDefault.metacardTypes.forEach(type -> {
+                        defaultAttributeValueRegistry.removeDefaultValue(type,
+                                theDefault.attribute);
+                    });
+                }
+            });
+            defaults.clear();
+        }
+
+        Map<String, Set<AttributeValidator>> validatorsByAttribute = attributeValidatorsByFile.get(
+                fileName);
+        if (validatorsByAttribute != null) {
+            validatorsByAttribute.forEach((attributeName, validatorsToRemove) -> {
+                Set<AttributeValidator> currentValidators =
+                        attributeValidatorRegistry.getValidators(attributeName);
+                Set<AttributeValidator> resultingValidators = difference(currentValidators,
+                        validatorsToRemove);
+                attributeValidatorRegistry.deregisterValidators(attributeName);
+
+                if (!resultingValidators.isEmpty()) {
+                    attributeValidatorRegistry.registerValidators(attributeName,
+                            resultingValidators);
+                }
+            });
+            validatorsByAttribute.clear();
+        }
+    }
+
+    private Set<AttributeValidator> difference(Set<AttributeValidator> original,
+            Set<AttributeValidator> toRemove) {
+        Set<AttributeValidator> result = new HashSet<>(original);
```

Comment:
Guava provides a `Sets.difference` that you could use instead of rolling your own.